CONTRIBUTING.md

```diff
@@ -0,0 +1,143 @@
+
+For some fairly complicated legal reasons we actually have three CLAs.  Which
+of the CLAs you sign depends upon whose behalf you are contributing:
+
+- All individuals (except employees of the U.S. government – see below) should
+  sign the [Individual Contributor License Agreement](http://www.projectcalico.org/community/Individual-Contributor-Agreement).
+- If you are contributing on behalf of a company or organization, *you still
+  need to sign the Individual Contributor License Agreement* as above, but
+  someone at your company or organization also needs to sign the
+  [Corporate Contributor License Agreement](http://www.projectcalico.org/community/corporate-contributor-agreement/)
+  providing a list of people authorized to commit code to Project Calico.
+- Employees of the U.S. government do not sign the Individual Contributor
+  License Agreement.  Instead, someone with authority to sign on behalf of your
+  agency should sign the [U.S Government Contributor License Agreement](http://www.projectcalico.org/community/us-government-contributor-agreement).
+
+It is important to reiterate: until you sign the required agreements we cannot
+accept your contribution!
+
+## Mailing lists and chat
+
+A great way to talk to us, ask questions, discuss features and bounce ideas
+around is to join one of the channels listed below:
+
+* [Technical Mailing List](http://lists.projectcalico.org/mailman/listinfo/calico-tech_lists.projectcalico.org)
+* [Slack Calico Users Channel](https://calicousers.slack.com) ([Sign up](https://calicousers-slackin.herokuapp.com))
+* IRC - [#calico](https://kiwiirc.com/client/irc.freenode.net/#calico)
+
+## Reporting issues
+
+Before raising an issue with *Libcalico*, please check for 
+duplicate issues, and read our [Troubleshooting](https://github.com/projectcalico/calico-containers/blob/master/docs/Troubleshooting.md)
+and our [Frequently Asked Questions](https://github.com/projectcalico/calico-containers/blob/master/docs/FAQ.md)
+documents.
+
+If you have a question, please hop on to our IRC or Slack channels (see above).
+
+If you do need to raise an issue, please include any of the following that may
+be relevant to your problem (including too much information is never
+a bad thing):
+
+-  A detailed description of the problem.
+-  The steps to reproduce the problem (e.g. the full list of `calicoctl`
+   commands that were run)
+-  The expected result and actual result.
+-  Versions of appropriate binaries and libraries.  For example, the output from
+   `calicoctl version`, your version of Docker, rkt, Kubernetes etc.
+-  A link to any diagnostics (e.g. if using `calicoctl`, you can gathered
+   diagnostics using `calicoctl diags` - this provides instructions for 
+   uploading the diags bundle to transfer.sh - or alternatively if the 
+   diagnostics contains sensitive information we can set up an alternative
+   method for transfer
+-  If using `calicoctl` the output from `calicoctl status` run on each node
```

Comment:
Indeed.  I think this section will need a rework.  I think you can assume people are using this as a Python library rather than part of a calico-containers deployment.